src/main/java/com/ericsson/gerrit/plugins/eiffel/messaging/EiffelEventSender.java

```diff
@@ -1,25 +1,121 @@
 import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.EiffelEventSenderException;
+import com.ericsson.eiffelcommons.utils.HttpRequest.HttpMethod;
 import com.google.gson.Gson;
-import com.google.gson.JsonObject;
 
 public class EiffelEventSender {
+    private static final String GENERATE_PUBLISH_ENDPOINT = "/generateAndPublish/";
+    private static final String MESSAGE_PROTOCOL = "mp";
+    private static final String MESSAGE_TYPE = "msgType";
+    private static final String EIFFEL_PROTOCOL = "eiffelsemantics";
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String APPLICATION_JSON = "application/json";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EiffelEventSender.class);
 
     private final Gson gson = new Gson();
-    private final EiffelPluginConfiguration pluginConfig;
+    private String eiffelMessage;
+    private String eiffelType;
+    private EiffelPluginConfiguration pluginConfig;
+    private HttpRequest httpRequest;
 
     public EiffelEventSender(EiffelPluginConfiguration pluginConfig) {
+        this.httpRequest = new HttpRequest(HttpMethod.POST);
         this.pluginConfig = pluginConfig;
     }
 
-    public void send(EiffelEvent eiffelEvent) {
-        JsonObject eiffelEventJson = gson.toJsonTree(eiffelEvent).getAsJsonObject();
-        System.out.println("Event to send::: " + eiffelEventJson.toString());
-        System.out.println("REMReM URL ::: " + pluginConfig.getRemremPublishURL());
-        // TODO Implement event sending using REMReM details from pluginconfig.
+    public EiffelEventSender(EiffelPluginConfiguration pluginConfig, HttpRequest httpRequest) {
+        this.httpRequest = httpRequest;
+        this.pluginConfig = pluginConfig;
+    }
+
+    /**
+     * Sends a REMReM eiffel message using
+     *
+     */
+    public void send() {
+        try {
+            generateAndPublish();
+        } catch (URISyntaxException | IOException | EiffelEventSenderException e) {
+            LOGGER.error("Failed to send eiffel message.", e);
+        }
+    }
+
+    public EiffelEventSender setMessage(final EiffelEvent eiffelMessage) {
+        this.eiffelMessage = gson.toJsonTree(eiffelMessage).toString();
+        return this;
+    }
+
+    public EiffelEventSender setType(final String eiffelType) {
+        this.eiffelType = eiffelType;
+        return this;
     }
 
+    private void generateAndPublish()
+            throws URISyntaxException, IOException, EiffelEventSenderException {
+        if (!isConfigurationSet()) {
+            final String errorMessage = String.format(
+                    "Neccessary configuration is missing to send event."
+                            + "\neiffelMessage: %s\neiffelType: %s\neiffelProtocol: %s",
+                    eiffelMessage, eiffelType, EIFFEL_PROTOCOL);
+            throw new EiffelEventSenderException(errorMessage);
+        }
```

Comment:
1. You could break this out to its own method.
2. Why check this earlier?

Reply:
It was in it's own method before but was later moved in here. It was moved in here at the request of someone else. I don't care either way though it would perhaps be more proper to break it out and place it in before you call generateAndPublish method. @e-edling-ericsson could perhaps be convinced that breaking it out is better after all.

Reply:
I broke it out and placed it before generateAndPublish

Reply:
Yes I think it's better to break it out to "verifyConfiguration" and put it right before "generateAndPublish"